fhir/scripts/populate.js

```diff
@@ -0,0 +1,149 @@
+const asyncHandler = require('../src/lib/async-handler');
+const { COLLECTION } = require('../src/constants');
+const { mongoConfig } = require('../src/config');
+const package = require('../package.json');
+const mongo = require('../src/lib/mongo');
+const program = require('commander');
+const path = require('path');
+const glob = require('glob');
+
+// const {
+// 	PatientSchema,
+// 	ObservationSchema
+// } = require('fhir-schemas');
+
+// const SCHEMA_MAP = {
+// 	'Patient': PatientSchema,
+// 	'Observation': ObservationSchema
+// };
+
+const VALID_PROFILE_NAMES = [
+	COLLECTION.OBSERVATION,
+	COLLECTION.PATIENT
+];
+
+/**
+ * @function loadProfiles
+ * @summary Load the profiles into Mongo and log status as we go
+ */
+let loadProfiles = async args => {
+	let { profiles, reset } = args;
+	// Connect to Mongo
+	let [ err, client ] = await asyncHandler(mongo(mongoConfig.connection, mongoConfig.options));
+
+	// Fail if ther is an error
+	if (err) {
+		console.error(err);
+		process.exit(err);
+	}
+
+	// Connect to our Database
+	let db = client.db(mongoConfig.db_name);
+	let collectionList = await db.listCollections().toArray();
+	let currentCollections = collectionList.map(coll => coll.name);
+
+	// Iterate over our profiles and insert  all of our documents
+	let profile_names = Object.keys(profiles);
+	for (let name of profile_names) {
+		let documents = profiles[name];
+
+		// Add a validator for the schemas
+		// let [ schemaErr, collection ] = await asyncHandler(db.createCollection(name, {
+		// 	validator: { $jsonSchema: SCHEMA_MAP[name] }
+		// }));
+    //
+		// // Skip this iteration if there was an error creating the schema
+		// if (schemaErr) {
+		// 	console.warn(`Unable to create collection for ${name} profile. Skipping.`);
+		// 	console.error(schemaErr);
+		// 	return;
+		// }
+		let collection = db.collection(name);
+
+		if (reset && currentCollections.indexOf(name) > -1) {
+			await asyncHandler(collection.drop());
+		}
+
+		// Insert all of our documents
+		console.log(`\nInserting documents for the ${name} profile.`);
+		let [insertErr, results] = await asyncHandler(collection.insertMany(documents));
+
+		if (insertErr) {
+			console.warn(`Unable to insert documents for ${name} profile.`);
+			console.error(insertErr);
+		}
+		else {
+			console.log(`Success. Inserted ${results.result.n} documents in ${name} profile.`);
+		}
+	}
+
+	// Close our connection
+	console.log('\nClosing connection\n');
+	client.close();
+};
+
+/**
+ * @function loadDocumentsForProfile
+ * @summary Search the fixtures directory and find matching files
+ */
+let loadDocumentsForProfile = profile_name => {
+	// Get a list of json files in fixtures
+	let basePath = path.join(__dirname, '../fixtures');
+	let profiles = glob.sync(path.join(basePath, '**/*.json'));
+	let modules = profiles
+		.map(profile_path => require(profile_path))
+		.filter(module => module.resourceType === profile_name);
```

Comment:
This seems really inefficient.  Wouldn't it be better to load them all into a map that you can reuse for each profile?

Reply:
You could but it won't not make a difference, node cache's modules when they are required and does not load them on future requires, it just pulls them from require.cache

Reply:
Right, ok.  I guess it's fine then